_source/_posts/2023-01-23-keycloak-kubernetes-prod.md

```diff
@@ -0,0 +1,637 @@
+---
+layout: blog_post
+title: "OAuth2 microservices with Keycloak on Google Kubernetes Engine"
+author:
+by: contractor
+communities: [devops,security,java]
+description: "A walk-through of building a microservices architecture with JHipster, Keycloak for OAuth2, and Google Kubernetes Engine (GKE) deployment. Keycloak as identity broker with Auth0 as the identity provider."
+tags: [keycloak, kubernetes, java, jhipster, security, oauth2]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: awareness
+---
+Keycloak is an open-source identity and access management solution that allows you to secure your applications and services with ease. JHipster, on the other hand, is a powerful development platform that provides developers with the tools they need to create modern, scalable, and robust web applications.
+
+Keycloak is the default OpenID Connect server configured with JHipster. Using Keycloak with JHipster is an excellent way to ensure that your web applications are secure and well-protected from unauthorized access. By integrating Keycloak with JHipster, you can easily implement authentication and authorization in your application, manage user identities, and provide secure access to resources.
+
+The JHipster Kubernetes generator creates all the necessary Kubernetes resources, such as deployments, services, and ingresses, based on the configuration of the JHipster application. This includes setting up the database, configuring security, and setting up any necessary environment variables. Since JHipster 8, the Kubernetes sub-generator supports Keycloak for Ingress GKE deployment. In this post, I'll walk you through the generation of a demo JHipster microservices application with Keycloak integration, and its deployment to Google Kubernetes Engine (GKE).
+
+{% img blog/keycloak-kubernetes-prod/keycloak-gke.png alt:"Keycloak, JHipster, GKE logos" width:"500" %}{: .center-image }
+
+{% include toc.md %}
+
+## Best practices for Keycloak in production
```

Comment:
I've also seen "Recommended" as a word instead of "Best".

Reply:
Updated